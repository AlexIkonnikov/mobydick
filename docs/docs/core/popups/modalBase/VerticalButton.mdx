import VerticalButtonsView from '../assets/VerticalButtonsView.png';

<img src={VerticalButtonsView} width="50%"/>

Reusable VerticalButton component for ModalBase

## Example

```tsx
import {
  Button,
  createStyles,
  ModalBase,
  Typography,
  usePopups,
  useStyles,
  View,
} from '@lad-tech/mobydick-core';

export const ModalWidget = () => {
  const [styles] = useStyles(styleFn);
  const {openPopup} = usePopups();

  return (
    <View style={styles.container}>
      <Typography font={'Regular-Primary-H5'}>Modal</Typography>
      <Button
        text={'ModalBase'}
        onPress={() =>
          openPopup({
            Content: props => (
              <ModalBase {...props}>
                <ModalBase.VerticalButtonsView>
                  <ModalBase.VerticalButton text={'VerticalButton'} />
                </ModalBase.VerticalButtonsView>
              </ModalBase>
            ),
          })
        }
      />
    </View>
  );
};

const styleFn = createStyles(({spaces}) => ({
  container: {
    gap: spaces.Space8,
  },
}));
```


## `Props`

### [`IButton Props`](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/cta/components/Button/types.ts#L35)

Inherits [IButton Props](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/cta/components/Button/types.ts#L35)



