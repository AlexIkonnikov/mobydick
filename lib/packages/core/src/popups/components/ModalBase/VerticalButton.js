"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const types_1 = require("../../../cta/components/Button/types");
const Button_1 = __importDefault(require("../../../cta/components/Button/Button"));
const VerticalButton = props => {
    return <Button_1.default size={types_1.IButtonSize.fixed} {...props}/>;
};
exports.default = VerticalButton;
