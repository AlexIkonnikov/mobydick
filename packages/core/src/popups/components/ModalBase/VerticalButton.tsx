import {FC} from 'react';

import {IButtonProps, IButtonSize} from '../../../cta/components/Button/types';
import Button from '../../../cta/components/Button/Button';

const VerticalButton: FC<IButtonProps> = props => {
  return <Button size={IButtonSize.fixed} {...props} />;
};

export default VerticalButton;
