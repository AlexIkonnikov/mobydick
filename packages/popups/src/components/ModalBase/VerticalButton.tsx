import React, {FC} from 'react';
import {Button, IButtonProps, ISize} from '@npm/mobydick-cta';

const VerticalButton: FC<IButtonProps> = props => {
  return <Button size={ISize.fixed} {...props} />;
};

export default VerticalButton;
