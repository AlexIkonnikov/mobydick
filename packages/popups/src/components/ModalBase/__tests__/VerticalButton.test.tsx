import {render} from '@testing-library/react-native';
import {IButtonTypes} from '@npm/mobydick-cta';
import React from 'react';

import VerticalButton from '../VerticalButton';

describe('@npm/mobydick-popups/modalBase', () => {
  it('should renders correctly VerticalButton one button ', () => {
    const {toJSON} = render(
      <VerticalButton
        type={IButtonTypes.destructive}
        onPress={() => console.log('onPress')}
        text={'text'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
