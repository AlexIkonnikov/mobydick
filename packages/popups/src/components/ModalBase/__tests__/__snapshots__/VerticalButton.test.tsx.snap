// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/modalBase should renders correctly VerticalButton one button  1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#F54D3D",
      "borderRadius": 18,
      "flexDirection": "row",
      "justifyContent": "center",
      "minHeight": 72,
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 27,
          "lineHeight": 33,
        },
        Array [
          Object {
            "color": "#FFF",
            "paddingHorizontal": 9,
          },
          undefined,
        ],
      ]
    }
  >
    text
  </Text>
</View>
`;
